.github/workflows/tests.yml

```diff
@@ -54,3 +56,30 @@ jobs:
         run: docker load < docker-image.tar
       - name: Lint and Type Checking
         run: docker run --rm --network=host --entrypoint="" azafea-tests pipenv run lint
+
+  push:
+    name: Upload the Docker image to Docker Hub
+    runs-on: ubuntu-latest
+    # We only want to publish a new image when the master branch changes, not
+    # for PRs and only if tests finish successfully.
+    needs: tests
+    if: github.event_name == 'push' || github.event_name == 'pull_request'
```

Comment:
Isn't this condition the wrong way round? To me this looks like it will run for pull requests, which is the opposite of what we want

Reply:
You're right, but instead of merging to `master` and only then would the job run (and then open another PR with another fix), I added that to the condition and marked the PR as a draft until it's working properly.

Reply:
Ah, I got it!